### PR TITLE
Use more descriptive task names for ILM cluster state updates

### LINKED ISSUE #34118: Enhance ILM cluster state update debuggability
Most of our ILM cluster state updates use `"ILM"` as the cluster state reason string, we should change this to a more descriptive string.

This will make debugability much easier when reading logs.

### PR DESCRIPTION
**Pull request by dakrone**
Rather than using "ILM" for everything, we should use more descriptive names so
debugging from logs is easier to do.

Resolves #34118

### CODE REVIEW

**Review by jasontedor — APPROVED** [review]
LGTM. I left one comment.

File: x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycleRunner.java [hunk 1]
```diff
@@ -383,18 +384,19 @@ static ClusterState addStepInfoToClusterState(Index index, ClusterState clusterS
     private void moveToStep(Index index, String policy, StepKey currentStepKey, StepKey nextStepKey) {
         logger.debug("moveToStep[" + policy + "] [" + index.getName() + "]" + currentStepKey + " -> "
             + nextStepKey);
-        clusterService.submitStateUpdateTask("ILM", new MoveToNextStepUpdateTask(index, policy, currentStepKey,
+        clusterService.submitStateUpdateTask("ILM-move-to-step", new MoveToNextStepUpdateTask(index, policy, currentStepKey,
```
[the remark this answers is not in the document]
Comment (dakrone) [reply x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycleRunner.java]: oh sure

**Review by AthenaEryma — APPROVED** [review]
LGTM2

File: x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycleRunner.java [hunk 2]
```diff
@@ -383,18 +384,19 @@ static ClusterState addStepInfoToClusterState(Index index, ClusterState clusterS
...
-        clusterService.submitStateUpdateTask("ILM", new MoveToNextStepUpdateTask(index, policy, currentStepKey,
+        clusterService.submitStateUpdateTask("ilm-move-to-step", new MoveToNextStepUpdateTask(index, policy, currentStepKey,
             nextStepKey, nowSupplier));
     }
 
     private void moveToErrorStep(Index index, String policy, StepKey currentStepKey, Exception e) {
         logger.error("policy [" + policy + "] for index [" + index.getName() + "] failed on step [" + currentStepKey
             + "]. Moving to ERROR step.", e);
-        clusterService.submitStateUpdateTask("ILM", new MoveToErrorStepUpdateTask(index, policy, currentStepKey, e, nowSupplier));
+        clusterService.submitStateUpdateTask("ilm-move-to-error",
```
Comment (jasontedor) [inline x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycleRunner.java]: `ilm-move-to-error` -> `ilm-move-to-error-step`

**Pull request merged**